File: 프로그래머스/1/176963. 추억 점수/추억 점수.kt
class Solution {
    fun solution(name: Array<String>, yearning: IntArray, photo: Array<Array<String>>): IntArray {
        val nameToScore = mutableMapOf<String, Int>()
        
        // 이름 → 점수 맵핑
        for (i in name.indices) {
            nameToScore[name[i]] = yearning[i]
        }

        // 각 사진별 점수 계산
        val result = mutableListOf<Int>()
        for (pic in photo) {
            var total = 0
            for (person in pic) {
                total += nameToScore[person] ?: 0 // 없으면 0점
            }
            result.add(total)
        }

        return result.toIntArray()
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const nameScore = {};
    
    // 이름 → 그리움 점수 매핑
    for (let i = 0; i < name.length; i++) {
        nameScore[name[i]] = yearning[i];
    }

    const answer = [];

    for (const pic of photo) {
        let total = 0;
        for (const person of pic) {
            total += nameScore[person] || 0; // 없으면 0점
        }
        answer.push(total);
    }

    return answer;
}
